pxr/imaging/hdSt/materialXFilter.cpp

```diff
@@ -293,6 +480,57 @@ _AddDefaultMtlxTextureValues(std::map<TfToken, VtValue>* hdTextureParams)
     (*hdTextureParams) [_tokens->sourceColorSpace] = VtValue(HdStTokens->raw);
 }
 
+static void
+_AddDefaultMtlxTextureValues(
+    mx::NodeDefPtr const& nodeDef,
+    std::map<TfToken, VtValue>* hdTextureParams)
+{
+    _AddDefaultMtlxTextureValues(hdTextureParams);
+
+    if (nodeDef->getCategory() == mx::ShaderNode::IMAGE) {
+        for (auto const& inputName: _textureParamTokens->allTokens) {
+            auto mxInput = nodeDef->getActiveInput(inputName.GetString());
+            if (mxInput && mxInput->hasValueString()) {
+                _GetHdTextureParameters(inputName,
+                                        mxInput->getValueString(),
+                                        hdTextureParams);
+            }
+        }
+    }
+
+    // Everything boils down to an <image> node. We might have to dig it out of
+    // the nodegraph. Unsure about triplanar that has 3 image nodes. Does Storm
+    // require per-image texture params? How does one specify that using a single
+    // token?
```

Comment:
Are you talking about sampler parameters (e.g. wrap mode)?  Hydra takes them specified per-texture, and in the texture metadata we're gathering we should have a per-texture entry for everything.  If the triplanar node only has one set of params, we can just splat that to all of the referenced textures.

Reply:
Yes, sample parameters could theoretically vary from one texture to the other. This is not the case with triplanar, but a custom NodeDef/NodeGraph node with multiple texture could try having one texture repeat while the other clamps. This is a bit of a corner case though, so unsure how much energy we should invest into covering it.

Reply:
I'm happy to leave it until we have a concrete asset we're trying to get working.